browser/BUILD.gn

```diff
@@ -190,8 +196,28 @@ source_set("browser_process") {
       "android/preferences/brave_pref_service_bridge.cc",
       "android/preferences/website/desktop_mode_tab_helper.cc",
       "android/preferences/website/desktop_mode_tab_helper.h",
+      "android/rewards/safetynet_check.cc",
+      "android/rewards/safetynet_check.h",
     ]
 
+    if (brave_rewards_enabled) {
+      sources += [
+        "android/rewards/brave_rewards_native_worker.cc",
```

Comment:
this should either be `brave_rewards/android/brave_rewards_native_worker.cc` or it should go in `src/brave/android/brave/rewards/brave_rewards_native_worker.cc`

Reply:
I guess there is already a lot of stuff in `src/brave/browser/android/...` and now that I search again that's consistent with chrome, but I think it should be `brave_rewards/android` instead of `android/rewards` so brave rewards code is grouped together

Reply:
I don't know as you see yourself all android related native files are inside `chrome/browser/android` in chromium. As far as we follow chromium's pattern in everything else I don't think we should do that.

Reply:
I guess we have to have both in that case `../../brave/components/brave_rewards/browser/rewards_service_private_observer.h:10:10: fatal error: 'bat/ledger/mojom_structs.h' file not found
#include "bat/ledger/mojom_structs.h"`

Reply:
that isn't actually correct. For instance `chrome/browser/history/android/...`, `chrome/browser/safe_browsing/android/...`, `chrome/browser/offline_pages/android/...`, etc... and we also want consistency with `rewards` -> `brave_rewards`

Reply:
There are definitely inconsistencies in the paths used in chrome, but grouping the files by component and then platform makes more sense to me and is more consistent with things like my_file.cc, my_file_android.cc, etc.. which would be in the same directory

Reply:
done